middlewares/userAuthorization.js

```diff
@@ -0,0 +1,28 @@
+const jwt = require("jsonwebtoken");
+const { User } = require("../models/user.model");
+const JWT_SECRET = process.env.JWT_SECRET;
+
+const userAuthorization = async (req, res, next) => {
+	try {
+		const tokenContainer = req.headers.authorization;
+		const token = tokenContainer.split(" ")[1];
+		const decoded = jwt.verify(token, JWT_SECRET);
+
+		const user = await User.findById(decoded.userId);
```

Comment:
It'll access the database on every api call which the user makes. I think you can store the entire user details except passwords and stuff inside the jwt token. and if the token is verified, then you don't need to check for the user in the database.